test/extended/kernel/common.go

```diff
@@ -0,0 +1,106 @@
+package kernel
+
+import (
+	"context"
+	"regexp"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+)
+
+var (
+	realTimeKernelRE = regexp.MustCompile(".*.rt[0-9]+.[0-9]+..*")
+	rtEnvFixture     = exutil.FixturePath("testdata", "kernel", "rt-tests-environment.yaml")
+	rtPodFixture     = exutil.FixturePath("testdata", "kernel", "rt-tests-pod.yaml")
+	rtNamespace      = "openshift-rt-tests"
+	rtPodName        = "rt-tests"
+)
+
+func skipIfNotRT(oc *exutil.CLI) {
+	g.By("checking kernel configuration")
+
+	cfg := oc.AdminConfig()
+	rt_nodes, err := getRealTimeWorkerNodes(cfg)
+	o.Expect(err).NotTo(o.HaveOccurred(), "unable to retrieve realtime worker nodes")
+
+	if len(rt_nodes) == 0 {
+		e2eskipper.Skipf("No realtime nodes detected")
+	}
+}
+
+func getRealTimeWorkerNodes(kubeClientConfig *rest.Config) (nodes map[string]int, err error) {
```

Comment:
nit: I think you should be able to pass the `oc` object and run something like below instead of creating the kubeclient instance.

```go
oc.AsAdmin().KubeClient().CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{LabelSelector: "node-role.kubernetes.io/worker="})
```